### PR TITLE
LightProbe: Add support for serialization/deserialization.

### PR DESCRIPTION
**Pull request by Mugen87**
After 21e3eab971f19ec62131bdc17a012b8f611bc101, serialization/deserialization is now easy to add.

### CODE REVIEW

File: src/lights/LightProbe.js [hunk 1]
```diff
@@ -26,7 +28,6 @@ LightProbe.prototype = Object.assign( Object.create( Light.prototype ), {
 		Light.prototype.copy.call( this, source );
 
 		this.sh.copy( source.sh );
-		this.intensity = source.intensity;
```
Comment (Mugen87) [inline src/lights/LightProbe.js]: `intensity` is already copied in `Light.copy()`.

File: src/loaders/ObjectLoader.js [hunk 2]
```diff
@@ -817,6 +819,13 @@ ObjectLoader.prototype = Object.assign( Object.create( Loader.prototype ), {
 
 				break;
 
+			case 'LightProbe':
+
+				var sh = new SphericalHarmonics3().fromArray( data.sh );
+				object = new LightProbe( sh, data.intensity );
```
Comment (donmccurdy) [inline src/loaders/ObjectLoader.js]: fwiw I'm putting toJSON and fromJSON on LightProbeVolume: 

https://github.com/mrdoob/three.js/blob/c704ff0ebe1a42d7f97f2cf2c22f2d14be50de67/examples/jsm/lights/LightProbeVolume.js#L345-L410

Comment (Mugen87) [reply src/loaders/ObjectLoader.js]: So you suggest to introduce `LightProbe.fromJSON()`, right?

Comment (donmccurdy) [reply src/loaders/ObjectLoader.js]: I'd suggest https://github.com/mrdoob/three.js/issues/11266 as a long-term goal. But I won't insist on this PR. :)

Comment (Mugen87) [reply src/loaders/ObjectLoader.js]: Okay, I've aligned the PR to yours. I think it makes sense if both follow the same approach 👍

**Comment by mrdoob** [comment]
> After 21e3eab971f19ec62131bdc17a012b8f611bc101, serialization/deserialization is now easy to add.

I forgot I did that. 👴

File: src/lights/LightProbe.js (hunk 1 above)
Comment (mrdoob) [reply src/lights/LightProbe.js]: As a FYI, seems like @WestLangley is thinking of removing that inheritance. https://github.com/mrdoob/three.js/pull/18371#issuecomment-603550543

File: src/lights/LightProbe.js [hunk 3]
```diff
@@ -36,7 +46,7 @@ LightProbe.prototype = Object.assign( Object.create( Light.prototype ), {
 
 		var data = Light.prototype.toJSON.call( this, meta );
 
-		// data.sh = this.sh.toArray(); // todo
+		data.object.sh = this.sh.toArray();
```
Comment (mrdoob) [inline src/lights/LightProbe.js]: `data.object.sh`?

Comment (Mugen87) [reply src/lights/LightProbe.js]: Check out `Light.toJSON()`:

https://github.com/mrdoob/three.js/blob/df2b31f86dbc82b4d852896c3f85f1a15f4b9f42/src/lights/Light.js#L44

File: src/lights/LightProbe.js (hunk 1 above)
Comment (Mugen87) [reply src/lights/LightProbe.js]: Thanks for sharing, good to know! Should I keep the code as it is now?

Comment (WestLangley) [reply src/lights/LightProbe.js]: I think so -- for now.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**